transactions/src/main/java/org/ehcache/transactions/xa/internal/xml/TxCacheServiceConfigurationParser.java

```diff
@@ -29,35 +28,30 @@
 
 import javax.xml.transform.Source;
 import javax.xml.transform.stream.StreamSource;
-import java.io.IOException;
 import java.net.URI;
-import java.net.URL;
+import java.util.Map;
+import java.util.function.Supplier;
 
+import static java.util.Collections.singletonMap;
 import static org.ehcache.transactions.xa.internal.xml.TxCacheManagerServiceConfigurationParser.TRANSACTION_NAMESPACE_PREFIX;
 
 /**
  * @author Ludovic Orban
  */
 @Component
-public class TxCacheServiceConfigurationParser extends BaseConfigParser<XAStoreConfiguration> implements CacheServiceConfigurationParser<XAStore.Provider> {
+public class TxCacheServiceConfigurationParser extends BaseConfigParser<XAStoreConfiguration> implements CacheServiceConfigurationParser<XAStore.Provider, XAStoreConfiguration> {
 
   private static final URI NAMESPACE = URI.create("http://www.ehcache.org/v3/tx");
-  private static final URL XML_SCHEMA = TxCacheManagerServiceConfigurationParser.class.getResource("/ehcache-tx-ext.xsd");
   private static final String STORE_ELEMENT_NAME = "xa-store";
   private static final String UNIQUE_RESOURCE_NAME = "unique-XAResource-id";
 
   @Override
-  public Source getXmlSchema() throws IOException {
-    return new StreamSource(XML_SCHEMA.openStream());
+  public Map<URI, Supplier<Source>> getSchema() {
+    return singletonMap(NAMESPACE, () -> new StreamSource(getClass().getResourceAsStream("/ehcache-tx-ext.xsd")));
```

Comment:
What happens if the schema isn't located?

Reply:
NPE at schema assembly.